src/PIL/Image.py

```diff
@@ -2236,20 +2236,19 @@ def thumbnail(self, size, resample=BICUBIC, reducing_gap=2.0):
         :returns: None
         """
 
-        # preserve aspect ratio
-        x, y = self.size
-        if x > size[0]:
-            y = max(round(y * size[0] / x), 1)
-            x = round(size[0])
-        if y > size[1]:
-            x = max(round(x * size[1] / y), 1)
-            y = round(size[1])
-        size = x, y
-        box = None
-
-        if size == self.size:
+        x, y = size
+        if x >= self.width and y >= self.height:
```

Comment:
Since the aspect ratio is being preserved I don't see how one side being smaller would make the image bigger. This does bring up something that I hadn't thought about: what happens when you pass in a size that's only fractionally smaller and it gets rounded up to the image's size?

Reply:
This will be handled by the next `if self.size != size:` statement, wouldn't?